src/helpers/env.js

```diff
@@ -73,78 +97,79 @@ function _getRuntimeEnv(runtime) {
-      type = mapping.type || type;
-    } else throw new Error("Unfamiliar mapping");
+    if (!name || !type) {
+      console.debug(runtime, "unnamed / untyped env mapping", key, mappedKey);
+      continue;
+    }
 
     if (runtime === "node") env[key] = process.env[name];
-    else if (runtime === "deno") env[key] = name && Deno.env.get(name);
+    else if (runtime === "deno") env[key] = Deno.env.get(name);
     else if (runtime === "worker") env[key] = globalThis[name];
-    else throw new Error(`Unknown runtime: ${runtime}`);
+    else throw new Error(`unsupported runtime: ${runtime}`);
 
-    // All env are assumed to be strings, so typecast them.
+    // env vars are strings by default, typecast to their respective types
     if (type === "boolean") env[key] = !!env[key];
     else if (type === "number") env[key] = Number(env[key]);
     else if (type === "string") env[key] = env[key] || "";
-    else throw new Error(`Unsupported type: ${type}`);
+    else throw new Error(`unsupported type: ${type}`);
+
+    console.debug("added", key, mappedKey, env[key]);
   }
 
   return env;
 }
 
 function _getRuntime() {
-  // As `process` also exists in worker, we need to check for worker first.
-  if (globalThis.RUNTIME === "worker") return "worker";
-  if (typeof Deno !== "undefined") return "deno";
-  if (typeof process !== "undefined") return "node";
+  if (typeof Deno !== "undefined") {
+    return "deno";
+  }
+
+  if (typeof process !== "undefined") {
+    // process also exists in Workers, where RUNTIME is defined
+    if (globalThis.RUNTIME) return globalThis.RUNTIME;
+    if (process.env) return process.env.RUNTIME || "node";
+  }
+
+  throw new Error("unknown runtime");
 }
 
 export default class EnvManager {
   /**
    * Initializes the env manager.
    */
   constructor() {
-    if (globalThis.env) throw new Error("envManager is already initialized.");
-
-    globalThis.env = {};
+    this.runtime = _getRuntime();
     this.envMap = new Map();
-    this.isLoaded = false;
+    this.load();
   }
 
   /**
    * Loads env variables from runtime env. and is made globally available
    * through `env` namespace. Existing env variables will be overwritten.
    */
-  loadEnv() {
-    const runtime = _getRuntime();
-    const env = _getRuntimeEnv(runtime);
-    for (const [key, value] of Object.entries(env)) {
-      this.envMap.set(key, value);
+  load() {
+    const renv = _getRuntimeEnv(this.runtime);
+
+    globalThis.env = renv; // Global `env` namespace.
+
+    for (const [k, v] of Object.entries(renv)) {
+      this.envMap.set(k, v);
     }
 
-    // adding download timeout with worker time to determine worker's overall
-    // timeout
-    runtime === "worker" &&
-      this.envMap.set(
-        "workerTimeout",
-        Number(WORKER_TIMEOUT) + Number(CF_BLOCKLIST_DOWNLOAD_TIMEOUT)
-      );
-
-    console.debug(
-      "Loaded env: ",
-      (runtime === "worker" && JSON.stringify(this.toObject())) ||
-        this.toObject()
```

Comment:
This provided a nice summary of env object loaded rather than `console.debug("added", key, mappedKey, env[key]);` above. Stringify is only needed on workers.

Reply:
Debug logs are better when verbose, rather than informative?

Reply:
I included `stringify` for all 3 runtimes, because conditionals just for Workers' `console.log` doesn't really mean much.